src/commands/devfile/generate.ts

```diff
@@ -0,0 +1,266 @@
+/*********************************************************************
+ * Copyright (c) 2019 Red Hat, Inc.
+ *
+ * This program and the accompanying materials are made
+ * available under the terms of the Eclipse Public License 2.0
+ * which is available at https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ **********************************************************************/
+// tslint:disable:object-curly-spacing
+
+import { V1beta1Ingress, V1Deployment, V1DeploymentSpec, V1ObjectMeta, V1PersistentVolumeClaim, V1PersistentVolumeClaimSpec, V1PodTemplateSpec, V1Service, V1ServicePort, V1ServiceSpec } from '@kubernetes/client-node'
+import { Command, flags } from '@oclif/command'
+import { string } from '@oclif/parser/lib/flags'
+import * as yaml from 'js-yaml'
+
+import { Devfile, DevfileComponent, DevfileProject, ProjectSource, TheEndpointName } from '../../api/devfile'
+import { KubeHelper } from '../../api/kube'
+
+const kube = new KubeHelper()
+const stringLitArray = <L extends string>(arr: L[]) => arr
+const languages = stringLitArray(['java', 'typescript', 'go', 'python', 'c#'])
+export type Language = (typeof languages)[number]
+
+const LanguagesComponents = new Map<Language, DevfileComponent>([
+  ['java', {type: TheEndpointName.ChePlugin, alias: 'java-ls', id: 'org.eclipse.che.vscode-redhat.java:0.38.0'}],
```

Comment:
@l0rd when are you going to merge this PR? If it lands after the new plugin notation https://github.com/eclipse/che/pull/13204 we'll need to update this PR

Reply:
@garagatyi hopefully by the beginning of the week. Anyway thanks for the heads up. I don't know what will be merged first but it should be pretty straightforward to fix but we need to think about it.

Reply:
I will work on this PR today and will update it with the new naming convention

Reply:
ok